aspnetcore/mvc/controllers/application-model.md

```diff
@@ -12,7 +12,187 @@ ms.technology: aspnet
+* [`IControllerModelConvention`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.icontrollermodelconvention)
+* [`IActionModelConvention`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.iactionmodelconvention)
+* [`IParameterModelConvention`](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.applicationmodels.iparametermodelconvention)
+
+Conventions are applied by adding them to MVC settings or by implementing `Attribute`s and applying them to controllers, actions, or action parameters (similar to [`Filters`](https://docs.microsoft.com/aspnet/core/mvc/controllers/filters)). Unlike filters, conventions are only executed when the app is starting, not as part of each request.
+
+### Sample: Modifying the ApplicationModel
+
+The following convention is used to add a property to the application model. 
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Conventions/ApplicationDescription.cs)]
+
+Application model conventions are applied as options when MVC is added in `ConfigureServices` in `Startup`.
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Startup.cs?name=ConfigureServices&highlight=5)]
+
+Properties are accessible from the `ActionDescriptor` properties collection within controller actions:
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Controllers/AppModelController.cs?name=AppModelController)]
+
+### Sample: Modifying the ControllerModel Description
+
+As in the previous example, the controller model can also be modified to include custom properties. These will override existing properties with the same name specified in the application model. The following convention attribute adds a description at the controller level:
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Conventions/ControllerDescriptionAttribute.cs)]
+
+This convention is applied as an attribute on a controller.
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Controllers/DescriptionAttributesController.cs?name=ControllerDescription&highlight=1)]
+
+The "description" property is accessed in the same manner as in previous examples.
+
+### Sample: Modifying the ActionModel Description
+
+A separate attribute convention can be applied to individual actions, overriding behavior already applied at the application or controller level.
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Conventions/ActionDescriptionAttribute.cs)]
+
+Applying this to an action within the previous example's controller demonstrates how it overrides the controller-level convention:
+
+[!code-csharp[Main](./application-model/sample/src/AppModelSample/Controllers/DescriptionAttributesController.cs?name=DescriptionAttributesController&highlight=9)]
+
+### Sample: Modifying the ParameterModel
```

Comment:
Ok, I'm just going to have it work as an attribute rather than adding it globally. It will still get the point across without exposing the lack of support for adding ParameterModelConvention globally. We can add an issue to come back to this and add it once aspnet/Mvc#5816 is complete.